internal/auth0/prompt/resource_screen_render.go

```diff
@@ -0,0 +1,210 @@
+				Description: "The screen that you are configuring settings for. " +
+					"Options are: `" + strings.Join(allowedScreensSettingsRenderer, "`, `") + "`.",
+			},
+			"rendering_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      standardMode,
+				ValidateFunc: validation.StringInSlice(supportedRenderingModes, false),
+				Description: "Rendering mode" +
+					"Options are: `" + strings.Join(supportedRenderingModes, "`, `") + "`.",
+			},
+			"tenant": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Tenant ID",
+			},
+			"context_configuration": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Context values to make available",
+			},
+			"default_head_tags_disabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Override Universal Login default head tags",
+			},
+			"head_tags": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ValidateFunc:     validation.StringIsJSON,
+				DiffSuppressFunc: structure.SuppressJsonDiff,
+				Description:      "An array of head tags",
+			},
+		},
+	}
+}
+
+func createPromptScreenRenderer(ctx context.Context, data *schema.ResourceData, meta any) diag.Diagnostics {
+	promptName := data.Get("prompt_type").(string)
+	screenName := data.Get("screen_name").(string)
+	data.SetId(fmt.Sprintf("%s:%s", promptName, screenName))
+	return updatePromptScreenRenderer(ctx, data, meta)
+}
+
+func readPromptScreenRenderer(ctx context.Context, data *schema.ResourceData, meta any) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	promptScreenSettings, err := api.Prompt.ReadRendering(ctx, management.PromptType(strings.Split(data.Id(), ":")[0]), management.ScreenName(strings.Split(data.Id(), ":")[1]))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return diag.FromErr(flattenPromptScreenSettings(data, promptScreenSettings))
+}
+
+func updatePromptScreenRenderer(ctx context.Context, data *schema.ResourceData, meta any) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	prompt := management.PromptType(data.Get("prompt_type").(string))
+	screen := management.ScreenName(data.Get("screen_name").(string))
+
+	promptSettings, err := expandPromptSettings(data)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := api.Prompt.UpdateRendering(ctx, prompt, screen, promptSettings); err != nil {
+		return diag.FromErr(internalError.HandleAPIError(data, err))
+	}
+
+	return readPromptScreenRenderer(ctx, data, meta)
+}
+
+func deletePromptScreenRenderer(ctx context.Context, data *schema.ResourceData, meta any) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	idComponents := strings.Split(data.Id(), ":")
+	promptName, screenName := idComponents[0], idComponents[1]
```

Comment:
Since in delete and update scenario, the id would have already created, and that id is being fetched from the data.
So, I hope no validations are required